### DISCUSSION TITLE
html email Template

### DISCUSSION THREAD

**Discussion by richard-tate-cf** [Accenture/sfmc-devtools#949] [Q&A]
we have a custom html template that we are trying to move from our Development Business unit to our QA Business unit it contains custom blocks that we have built using this https://www.npmjs.com/package/sfmc-sdk package.
The blocks are being moved over but the html markup is not, even though html file is showing in git and we can see the the file in bitbucket diff. I had noticed that template files where added to .gitignore file, but I removed this line for testing and wanted to know if html can be moved from one business unit to an other using your tool?

**Comment by JoernBerkefeld** [comment]
uhm... sfmc-sdk is a generic wrapper around SFMC's APIs. mcdev is using that under the hood. Not sure I understand what you were doing with it.

- how exactly are you doing the "moving"? could you please detail the steps that you were following, including commands you executed?
- Copying email templates (asset/message folder) is definitely supported. If it didn't work you either found a bug or did it wrong somehow.
- the template/ folder is excluded from git by default because it is meant to be used as a temporary folder - similar to the deploy/ folder. That said, there are projects in which I've chosen to commit it to git myself.

**Reply by richard-tate-cf** [reply]
Thanks for your response, We are creating dynamic email block components for our Marketing team, which uses a base custom template that holds all the css. The package we are using in the npm third party package called blocksdk, sorry for confusion on that one. As for how we are moving we developing it in our dev business unit and once its ready for testing we move it using your tool using the following command mcdev retrieve Devops/DEV , which gets all changes and then we select the changes we want to push to our testing repo, so we can have version controlling in our Business units. but it does sound like it's a bug.

**Reply by JoernBerkefeld** [reply]
ok, looked it up: https://www.npmjs.com/package/blocksdk
That's a wrapper around Content Builder API if I'm not mistaken. Haven't used it TBH but it should not cause any issues with mcdev or the GUI.

Could you maybe create a small demo for me @richard-tate-cf ? In essence, I would want you to create a simple content block with some HTML in it that is lost during your copy attempt. If you want, you can also share one that already exists as long as that doesn't contain any confidential / client info. 

Also, when you try and RETRIEVE that content block (please specify type "asset" and the key of the content block while executing retrieve!) do please append `--api=log` to it and then share that log file with me. It should create 2 files, one ending on error.log and one without "error" in the name. That second one is what I need.

The command should look something like:
`mcdev retrieve Devops/DEV asset yourContenBlock'sKeyHere --api=log`


With the log and the resulting asset (might be a folder!) I can try to reproduce this on our server and see what I can do about it.
